src/Sentry/Protocol/Envelopes/Envelope.cs

```diff
@@ -73,18 +74,24 @@ public async Task SerializeAsync(Stream stream, CancellationToken cancellationTo
         /// <summary>
         /// Creates an envelope that contains a single event.
         /// </summary>
-        public static Envelope FromEvent(SentryEvent @event)
+        // TODO: use builders for this?
```

Comment:
Can't use a builder because there are many ways this would be built wrong.

```suggestion
```

Can't have 2 events in. can't have 1 event and 1 transaction. Can have many sessions.
Can have stand alone attachment, but must include event id in the envelope header.

I'd continue with some fixed public methods that create it with the right/valid use cases. For now.

Reply:
How is a standalone attachment sent?

Reply:
@Tyrrrz you need to give a event id at least

Reply:
No, I mean from the SDK perspective. I didn't see `Sentry.CaptureAttachment` in any other SDK.

Reply:
@Tyrrrz I don't think we have it, the idea is to only send an attachments to an event. So:

`CaptureEvent(@event, s => s.AddAttachment(...))`

Although we still don't have the overload to take a `Action<Scope>`.

Raised it: https://github.com/getsentry/sentry-dotnet/issues/675